src/pyk/proof/exploration.py

```diff
@@ -0,0 +1,359 @@
+        for nid in pruned_nodes:
+            self._terminal_node_ids.discard(self.kcfg._resolve(nid))
+        return pruned_nodes
+
+    def shortest_path_to(self, node_id: NodeIdLike) -> tuple[KCFG.Successor, ...]:
+        spb = self.kcfg.shortest_path_between(self.init, node_id)
+        assert spb is not None
+        return spb
+
+    @staticmethod
+    def read_proof(id: str, proof_dir: Path) -> ExplorationProof:
+        proof_path = proof_dir / f'{hash_str(id)}.json'
+        if ExplorationProof.proof_exists(id, proof_dir):
+            proof_dict = json.loads(proof_path.read_text())
+            _LOGGER.info(f'Reading ExplorationProof from file {id}: {proof_path}')
+            return ExplorationProof.from_dict(proof_dict, proof_dir=proof_dir)
+        raise ValueError(f'Could not load ExplorationProof from file {id}: {proof_path}')
+
+    @property
+    def status(self) -> ProofStatus:
+        if len(self.kcfg.stuck) > 0:
+            return ProofStatus.FAILED
+        elif len(self.pending) > 0:
+            return ProofStatus.PENDING
+        else:
+            return ProofStatus.PASSED
+
+    @classmethod
+    def from_dict(
+        cls: type[ExplorationProof], dct: Mapping[str, Any], proof_dir: Path | None = None
+    ) -> ExplorationProof:
+        cfg = KCFG.from_dict(dct['cfg'])
+        init_node = dct['init']
+        terminal_node_ids = dct['terminal_node_ids']
+        id = dct['id']
+        if 'logs' in dct:
+            logs = {k: tuple(LogEntry.from_dict(l) for l in ls) for k, ls in dct['logs'].items()}
+        else:
+            logs = {}
+
+        return ExplorationProof(
+            id,
+            cfg,
+            init_node,
+            logs=logs,
+            terminal_node_ids=terminal_node_ids,
+            proof_dir=proof_dir,
+        )
+
+    def path_constraints(self, final_node_id: NodeIdLike) -> KInner:
+        path = self.shortest_path_to(final_node_id)
+        curr_constraint: KInner = mlTop()
+        for edge in reversed(path):
+            if type(edge) is KCFG.Split:
+                assert len(edge.targets) == 1
+                csubst = edge.splits[edge.targets[0].id]
+                curr_constraint = mlAnd([csubst.subst.ml_pred, csubst.constraint, curr_constraint])
+            if type(edge) is KCFG.Cover:
+                curr_constraint = mlAnd([edge.csubst.constraint, edge.csubst.subst.apply(curr_constraint)])
+        return mlAnd(flatten_label('#And', curr_constraint))
+
+    @property
+    def dict(self) -> dict[str, Any]:
+        dct = super().dict
+        dct['type'] = 'ExplorationProof'
+        dct['cfg'] = self.kcfg.to_dict()
+        dct['init'] = self.init
+        dct['terminal_node_ids'] = self._terminal_node_ids
+        logs = {k: [l.to_dict() for l in ls] for k, ls in self.logs.items()}
+        dct['logs'] = logs
+        return dct
+
+    @property
+    def summary(self) -> CompositeSummary:
```

Comment:
```suggestion
    def summary(self) -> ProofSummary:
```

I guess this should match the superclass type signature?

Reply:
I think so, if no subproofs are allowed.